app/Model/AppModel.php

```diff
@@ -1340,43 +1332,54 @@ public function updateDatabase($command, $useWorker=false)
                     ));
                     $this->__setUpdateResMessages($i, __('Successfuly executed the SQL query for ') . $command);
                 } catch (Exception $e) {
+                    $error_message = $e->getMessage();
                     $this->Log->create();
-                    $this->Log->save(array(
+                    $log_message = array(
                         'org' => 'SYSTEM',
                         'model' => 'Server',
                         'model_id' => 0,
                         'email' => 'SYSTEM',
                         'action' => 'update_database',
                         'user_id' => 0,
                         'title' => __('Issues executing the SQL query for ') . $command,
-                        'change' => __('The executed SQL query was: ') . $sql . PHP_EOL . __(' The returned error is: ') . $e->getMessage()
-                    ));
-                    $this->__setUpdateResMessages($i, __('Issues executing the SQL query for ') . $command . __('. The returned error is: ') . PHP_EOL . $e->getMessage());
-                    $this->__setUpdateError($i);
-                    $error_count++;
-                    if ($exitOnError) {
-                        $flag_stop = true;
-                        break;
+                        'change' => __('The executed SQL query was: ') . $sql . PHP_EOL . __(' The returned error is: ') . $error_message
+                    );
+                    $this->__setUpdateResMessages($i, __('Issues executing the SQL query for ') . $command . __('. The returned error is: ') . PHP_EOL . $error_message);
+                    $error_duplicate_column = 'SQLSTATE[42S21]: Column already exists: 1060 Duplicate column name';
+                    $error_duplicate_index = 'SQLSTATE[42000]: Syntax error or access violation: 1061 Duplicate key name';
```

Comment:
The only issue I can see with this sort of error handling is that we're tying ourselves even closer to mysql. Perhaps something more generic would be handy? Simple substring search for duplicate key name / duplicate column name?

Reply:
Just had a look at [mysql-error-reference](https://dev.mysql.com/doc/refman/8.0/en/server-error-reference.html). Doing a substring search for ``Duplicate column name`` works but ``Duplicate key`` is also present in other *mysql* error messages.
Moreover for *posgresql*, the error message for duplicate column is ``ERROR: column "x" specified more than once``, so a substring might not work.
What I can do is also collect and whitelist errors from *posgresql*.
What do you think?